### PR TITLE
Docker dev mode

### PR DESCRIPTION
**Pull request by francbartoli**
- [x] It enables running GeoNode for development with Docker
- [x] It requires a bit of documentation on the new-docs branch how to exploit it for debugging with ipdb https://github.com/GeoNode/geonode/commit/5a595ee37c0995fde601be96ba21e2a08269976b 
- [x] It allows to run development mode from the main docker-compose file

### CODE REVIEW

**Comment by francbartoli** [comment]
@afabiani if you can approve this contains also the fix for travis

**Comment by francbartoli** [comment]
Also it closes #4516

File: geonode/geoserver/views.py [hunk 1]
```diff
@@ -692,7 +692,7 @@ def get_layer_capabilities(layer, version='1.3.0', access_token=None, tolerant=F
     if not getattr(settings, 'DELAYED_SECURITY_SIGNALS', False):
         if tolerant and ('ServiceException' in getcap or req.status_code == 404):
             # WARNING Please make sure to have enabled DJANGO CACHE as per
-            # https://docs.djangoproject.com/en/2.0/topics/cache/#filesystem-caching
+            # https://docs.djangoproject.com/en/2.10/topics/cache/#filesystem-caching
```
Comment (frafra) [inline geonode/geoserver/views.py]: Broken links.

**Comment by frafra** [comment]
Wouldn't be better to have just a single `.env` file for development and one compose file just to override the main one?

**Comment by francbartoli** [comment]
@frafra do you mean just one single .env file loaded in each container?

**Comment by frafra** [comment]
@francbartoli I see that we already have different env file for different containers, so we can keep it like that; I was just trying to see if we can avoid duplication and redundancies.
We could also use a newer version for the compose file, but a new issue/PR would be probably better.

**Comment by francbartoli** [comment]
@frafra I will try to follow your suggestion to override from the main docker-compose script however

**Comment by afabiani** [comment]
@francbartoli so there's still more work to do on that as far as I understood, or are you planning to do additional alignments as part of different issues?

**Comment by francbartoli** [comment]
@afabiani yes I'll follow up with a different issue/PR to upgrade the version of docker-compose. For now, let's wait a bit so I can update this PR to use the main `docker-compose.yml` also for development

**Comment by afabiani** [comment]
@francbartoli any chance to close this before tomorrow?

**Comment by francbartoli** [comment]
@afabiani I will try to devote some spare time today in the late afternoon

**Comment by francbartoli** [comment]
@afabiani it is ready to merge, I will squash it after your review

**Comment by afabiani** [comment]
@francbartoli approved, seems good to me

**Pull request merged**